runtime-light/stdlib/variable-handling.cpp

```diff
@@ -49,6 +49,11 @@ void do_print_r(const mixed &v, int depth) {
       *coub << shift << ")\n";
       break;
     }
+    case mixed::type::OBJECT: {
+      php_warning("print_r used on object");
+      *coub << v.as_object()->get_class();
+      break;
+    }
     default:
       __builtin_unreachable();
```

Comment:
It's up to you, but why do we need a source of UB here? Isn't it better to replace `__builtin_unreachable` with, for example, `php_critical_error("non-exhaustive visitor")`?